membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -115,24 +114,34 @@ class AttributeController extends Controller with LoggingWithLogstashFields {
     def queryToAccountIds(response: QueryResponse): List[AccountId] =  response.records.map(_.Id)
 
     def getSubscriptions(accountIds: List[AccountId]): Future[Disjunction[String, List[Subscription[AnyPlan]]]] = {
-      def sub(accountId: AccountId): Future[Disjunction[String, List[Subscription[AnyPlan]]]] =
-        subscriptionService.subscriptionsForAccountId[AnyPlan](accountId)(anyPlanReads)
-
-      val maybeSubs: Future[Disjunction[String, List[Subscription[AnyPlan]]]] = accountIds.traverseU(id => sub(id)).map(_.sequenceU.map(_.flatten))
-
-      maybeSubs.map {
-        _.leftMap { errorMsg =>
-          log.error(s"We tried getting a subscription for a user with identityId ${identityId}, but then ${errorMsg}")
-          errorMsg
-        } map { subs =>
-          log.info(s"We got subs for identityId ${identityId} from Zuora and there were ${subs.length}")
-          subs
+      if(accountIds.isEmpty) Future.successful(\/.left(s"Can't get subscriptions for $identityId because they have no accountIds"))
+      else {
+        def sub(accountId: AccountId): Future[Disjunction[String, List[Subscription[AnyPlan]]]] =
+          subscriptionService.subscriptionsForAccountId[AnyPlan](accountId)(anyPlanReads)
+
+        val maybeSubs: Future[Disjunction[String, List[Subscription[AnyPlan]]]] = accountIds.traverseU(id => sub(id)).map(_.sequenceU.map(_.flatten))
+
+        maybeSubs.map {
+          _.leftMap { errorMsg =>
+            log.error(s"We tried getting a subscription for a user with identityId ${identityId}, but then ${errorMsg}")
+            errorMsg
+          } map { subs =>
+            log.info(s"We got subs for identityId ${identityId} from Zuora and there were ${subs.length}")
+            subs
+          }
         }
       }
     }
 
-    val attributes: DisjunctionT[Future, String, Option[Attributes]] = for {
-      accounts <- EitherT(withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraRestService.getAccounts(identityId)))
+    def zuoraAccountsQuery(identityId: String): Future[Disjunction[String, QueryResponse]] = zuoraRestService.getAccounts(identityId).map {
+      _.leftMap { error =>
+        log.error(s"Calling ZuoraAccountIdsFromIdentityId failed for identityId ${identityId}. Error: ${error}")
+        error
+      }
+    }
+
+    val attributes = for {
+      accounts <- EitherT[Future, String, QueryResponse](withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraAccountsQuery(identityId)))
       accountIds = queryToAccountIds(accounts)
       subscriptions <- EitherT[Future, String, List[Subscription[AnyPlan]]](withTimer(s"ZuoraGetSubscriptions", getSubscriptions(accountIds)))
```

Comment:
if an empty list is not acceptable for getSubscriptions, perhaps it would be better to change the type of the parameter to a scalaz.NonEmptyList just so it's clearer to anyone reusing it.  I'm not sure I'd expect this behaviour otherwise.

if on the other hand, getsubscriptions is using an either to signal back to the calling code whether it should be timing it, I think it's not really down to getsubscriptions to make that decision, it's better up to the thing doing the timing to work what it wants to time.
also it would need to be an either of either, as we want to know if something has failed, so we still need the "$whichCall failed with <error>"

all the extra 0 or 1 ms in the total may not be affecting the figures too much in reality, when the actual calls are adding ~250ms each time, unless we're getting (say) 95+% of calls  with nothing in the list?